src/scan/std_impls.rs

```diff
@@ -42,6 +42,29 @@ unsafe impl<K: Scan, V: Scan, S: BuildHasher> Scan for HashMap<K, V, S> {
 // FIXME: Would a bad build hasher cause problems?
 unsafe impl<K: GcSafe, V: GcSafe, S: BuildHasher> GcSafe for HashMap<K, V, S> {}
 
+unsafe impl<T: Scan> Scan for BTreeSet<T> {
+    #[inline]
+    fn scan(&self, scanner: &mut Scanner<'_>) {
+        for e in self {
+            scanner.scan(e)
+        }
+    }
+}
+
+unsafe impl<T: GcSafe> GcSafe for BTreeSet<T> {}
+
+unsafe impl<K: Scan, V: Scan> Scan for BTreeMap<K, V> {
+    #[inline]
+    fn scan(&self, scanner: &mut Scanner<'_>) {
+        for (k, v) in self {
+            scanner.scan(k);
+            scanner.scan(v);
+        }
+    }
+}
+
+unsafe impl<K: GcSafe, V: GcSafe> GcSafe for BTreeMap<K, V> {}
+
```

Comment:
Rebase off master

Reply:
Had a little bit of an issue getting git to recognize upstream/master vs. origin/master but I think I've solved that - let me know if it's not. It did involve a `git push --force` but I think it was safe, I wasn't overwriting commits.